src/libdredd/src/mutate_visitor.cc

```diff
@@ -77,10 +83,43 @@ bool MutateVisitor::TraverseDecl(clang::Decl* decl) {
   return true;
 }
 
+bool MutateVisitor::VisitUnaryOperator(clang::UnaryOperator* unary_operator) {
+  if (NotInFunction()) {
+    // Only consider mutating unary expressions that occur inside functions.
+    return true;
+  }
+
+  // Check that the unary expression and its arguments have source
+  // ranges that are part of the main file. In particular, this avoids
+  // mutating expressions that directly involve the use of macros (though it
+  // is OK if sub-expressions of arguments use macros).
+  if (GetSourceRangeInMainFile(compiler_instance_.getPreprocessor(),
+                               *unary_operator)
+          .isInvalid() ||
+      GetSourceRangeInMainFile(compiler_instance_.getPreprocessor(),
+                               *unary_operator->getSubExpr())
+          .isInvalid()) {
+    return true;
+  }
+
+  // We only want to change operators for unary operations on basic types.
+  // In particular, we do not want to mess with pointer arithmetic.
```

Comment:
I suppose for unary operators, pointer arithmetic isn't relevant, so I would remove this comment. (Or replace it with one about us not wanting to mess with the address-of and dereference unary operators, though it's not clear what we would do with those in any case.)

Reply:
Done.